core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,348 @@
+                    ' received %s' % type(self.assigned_reviewer_id))
+
+        if not isinstance(self.final_reviewer_id, basestring):
+            if self.final_reviewer_id:
+                raise utils.ValidationError(
+                    'Expected final_reviewer_id to be a string, received %s' %
+                    type(self.final_reviewer_id))
+
+        if not isinstance(self.score_category, basestring):
+            raise utils.ValidationError(
+                'Expected score_category to be a string, received %s' % type(
+                    self.score_category))
+
+        if (
+                suggestion_models.SCORE_CATEGORY_DELIMITER not in
+                self.score_category):
+            raise utils.ValidationError(
+                'Expected score_category to be of the form'
+                ' score_type%sscore_sub_type, received %s' % (
+                    suggestion_models.SCORE_CATEGORY_DELIMITER,
+                    self.score_category))
+
+        if (
+                len(self.score_category.split(
+                    suggestion_models.SCORE_CATEGORY_DELIMITER))) != 2:
+            raise utils.ValidationError(
+                'Expected score_category to be of the form'
+                ' score_type%sscore_sub_type, received %s' % (
+                    suggestion_models.SCORE_CATEGORY_DELIMITER,
+                    self.score_category))
+
+        if self.get_score_type() not in suggestion_models.SCORE_TYPE_CHOICES:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be among allowed'
+                ' choices, received %s' % self.get_score_type())
+
+    def accept(self):
+        """Accepts the suggestion. Each subclass must implement this function.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement accept.')
+
+    def get_change_list_for_accepting_suggestion(self):
+        """Before accepting the suggestion, a change_list needs to be generated
+        from the change_cmd. Each subclass must implement this function.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement '
+            'get_change_list_for_accepting_suggestion.')
+
+    def pre_accept_validate(self):
+        """Performs referential validation. This function needs to be called
+        before accepting the suggestion.
+        """
+        raise NotImplementedError(
+            'Subclasses of BaseSuggestion should implement '
+            'pre_accept_validate.')
+
+    @property
+    def is_handled(self):
+        """Returns if the suggestion has either been accepted or rejected.
+
+        Returns:
+            bool. Whether the suggestion has been handled or not.
+        """
+        return (self.status not in [suggestion_models.STATUS_IN_REVIEW,
+                                    suggestion_models.STATUS_RECEIVED])
```

Comment:
What does STATUS_RECEIVED represent? Does it refer to the scenario where suggestion has been received, but no reviewer has been assigned and hence, it can't be said to be in review?

Reply:
Yes